app/models/reading.rb

```diff
@@ -25,27 +25,34 @@ def self.mark_to_read_later(work, user, toread)
 
   # called from rake
   def self.update_or_create_in_database
-    REDIS_GENERAL.smembers("Reading:new").reverse.each_slice(ArchiveConfig.READING_BATCHSIZE || 1000) do |batch|
-      Reading.transaction do
-        batch.each do |reading_json|
-          Reading.reading_object(reading_json)
-        end
-      end
-    end
+    return unless REDIS_GENERAL.exists("Reading:new")
+
+    # Get a unique key associated with this run:
+    counter = REDIS_GENERAL.incrby("Reading:job_id", 1)
```

Comment:
Is this the same as `.incr("Reading:job_id")`?

Reply:
Yep, good catch!